compose/src/main/java/com/example/compose/component/SoptTextField.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.compose.component
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.compose.ui.theme.INSOPTAndroidPracticeTheme
+
+@Composable
+fun SoptTextField(
+    text: String,
+    hint: String,
+    singleLine: Boolean = true,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    keyboardActions: KeyboardActions = KeyboardActions.Default,
+    visualTransformation: VisualTransformation = VisualTransformation.None,
+    writeText: (String) -> Unit
```

Comment:
```suggestion
    onTextChange: (String) -> Unit
```

TextChangedListener니까....이런식으로 Event 발생에 대한 콜백느낌으로 만드는게 좋을 것 같은데 어떻게 생각해?

Reply:
네이밍은 어려워